workbase/src/renderer/components/shared/KeyspacesHandler.vue

```diff
@@ -77,55 +88,55 @@
 
 <script>
 
-import * as React from 'react';
-import { Button } from '@blueprintjs/core';
 import storage from '@/components/shared/PersistentStorage';
+
 import { mapGetters } from 'vuex';
 
 import { CURRENT_KEYSPACE_CHANGED } from './StoresActions';
+import ToolTip from '../UIElements/ToolTip';
 
 export default {
   name: 'KeyspacesList',
   props: ['showKeyspaceTooltip'],
+  components: { ToolTip },
   data() {
     return {
       keyspaceItems: [],
-      keyBtn: null,
       showKeyspaceList: false,
-      keyspaceBtn: null,
       clickEvent: () => {
         this.showKeyspaceList = false;
       },
+      btnText: 'keyspace',
     };
   },
   computed: {
     ...mapGetters(['allKeyspaces', 'currentKeyspace', 'isGraknRunning']),
   },
-  created() {
-    this.renderButton();
-  },
   watch: {
     allKeyspaces(val) {
       // If user deletes current keyspace from Keyspaces page, set new current keyspace to null
       if (!val.includes(this.currentKeyspace)) { this.$store.dispatch(CURRENT_KEYSPACE_CHANGED, null); }
     },
     isGraknRunning(val) {
       if (!val) {
-        this.$notifyError('It was not possible to retrieve keyspaces <br> - make sure Grakn is running <br> - check that host and port in connection settings are correct');
+        this.$notifyInfo('It was not possible to retrieve keyspaces <br> - make sure Grakn is running <br> - check that host and port in connection settings are correct');
       }
     },
-    currentKeyspace() {
-      this.renderButton();
-    },
-    showKeyspaceTooltip() {
-      this.renderButton();
+    currentKeyspace(ks) {
+      if (ks !== null) {
+        if (ks.length > 15) { // truncate long keyspace names
+          this.btnText = `${ks.substring(0, 15)}...`;
+        } else {
+          this.btnText = ks;
+        }
+      } else {
+        this.btnText = 'keyspace';
```

Comment:
I guess the checks and string manipulations belong to filters

https://vuejs.org/v2/guide/filters.html